ports/espressif/bindings/esp32_camera/Camera.c

```diff
@@ -0,0 +1,1017 @@
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_frame_size, MP_ARG_OBJ, {.u_obj = MP_ROM_NONE} },
+        { MP_QSTR_pixel_format, MP_ARG_OBJ, {.u_obj = MP_ROM_NONE} },
+        { MP_QSTR_grab_mode, MP_ARG_OBJ, {.u_obj = MP_ROM_NONE} },
+        { MP_QSTR_framebuffer_count, MP_ARG_OBJ, {.u_obj = MP_ROM_NONE} },
+    };
+
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args - 1, pos_args + 1, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    framesize_t frame_size =
+        args[ARG_frame_size].u_obj != MP_ROM_NONE
+        ?  validate_frame_size(args[ARG_frame_size].u_obj, MP_QSTR_frame_size)
+        : common_hal_esp32_camera_camera_get_frame_size(self);
+    pixformat_t pixel_format =
+        args[ARG_pixel_format].u_obj != MP_ROM_NONE
+        ?  validate_pixel_format(args[ARG_pixel_format].u_obj, MP_QSTR_pixel_format)
+        : common_hal_esp32_camera_camera_get_pixel_format(self);
+    camera_grab_mode_t grab_mode =
+        args[ARG_grab_mode].u_obj != MP_ROM_NONE
+        ?  validate_grab_mode(args[ARG_grab_mode].u_obj, MP_QSTR_grab_mode)
+        : common_hal_esp32_camera_camera_get_grab_mode(self);
+    bool framebuffer_count =
+        args[ARG_framebuffer_count].u_obj != MP_ROM_NONE
+        ?  mp_obj_get_int(args[ARG_framebuffer_count].u_obj)
+        : common_hal_esp32_camera_camera_get_framebuffer_count(self);
+
+    common_hal_esp32_camera_camera_reconfigure(self, frame_size, pixel_format, grab_mode, framebuffer_count);
+
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_KW(esp32_camera_camera_reconfigure_obj, 1, esp32_camera_camera_reconfigure);
+
+//|     pixel_format: PixelFormat
+//|     """The pixel format of captured frames"""
+
+STATIC mp_obj_t esp32_camera_camera_get_pixel_format(const mp_obj_t self_in) {
+    esp32_camera_camera_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+    return cp_enum_find(&esp32_camera_pixel_format_type, common_hal_esp32_camera_camera_get_pixel_format(self));
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(esp32_camera_camera_get_pixel_format_obj, esp32_camera_camera_get_pixel_format);
+
+MP_PROPERTY_GETTER(esp32_camera_camera_pixel_format_obj,
+    (mp_obj_t)&esp32_camera_camera_get_pixel_format_obj);
+
+
+//|     frame_size: FrameSize
+//|     """The size of captured frames"""
+
+STATIC mp_obj_t esp32_camera_camera_get_frame_size(const mp_obj_t self_in) {
+    esp32_camera_camera_obj_t *self = MP_OBJ_TO_PTR(self_in);
+    check_for_deinit(self);
+    return cp_enum_find(&esp32_camera_frame_size_type, common_hal_esp32_camera_camera_get_frame_size(self));
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(esp32_camera_camera_get_frame_size_obj, esp32_camera_camera_get_frame_size);
+
+MP_PROPERTY_GETTER(esp32_camera_camera_frame_size_obj,
+    (mp_obj_t)&esp32_camera_camera_get_frame_size_obj);
+
+//|     contrast: int
+//|     """Access the contrast property of the camera sensor"""
```

Comment:
I know it's annoying but these properties should really explain what the property is. That way folks don't need to look further for that info. If a property is unlikely to be used, then just remove it. It is ok to not expose everything.

Reply:
yes, agreed.